lib/task-graph.js

```diff
@@ -61,13 +61,57 @@ function taskGraphFactory(
         };
         if (this.context.target) {
             this.logContext.id = this.context.target;
-        }
+       }
         // For database ref linking
         this.node = this.context.target || this.definition.options.nodeId;
 
         return this;
     }
 
+    /**
+    This is a helper that visits one node. 
+    This function is used in _visitGraphNode().
+    */
+
+    TaskGraph.prototype._visitOneGraphNode = function(task, value, dep) {
```

Comment:
I agree w/ @yyscamper; it's a little confusing especially because of the _visitGraphNode recursion. Something to the effect of 'do-stuff-with / get NonterminalStates'.